community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v3_2/codegen/AuxGenerator.scala

```diff
@@ -75,14 +82,177 @@ class AuxGenerator(val packageName: String, val generator: CodeGenerator) {
     })
   }
 
-  private def newValueTypeName() = {
-    val name = "ValueType" + nameId
+  private def sortResultSign(sortOrder: SortOrder): Int = sortOrder match {
+    case Ascending => -1
+    case Descending => +1
```

Comment:
Isn't it more convenient to embed these constants with the `SortOrder` objects, and then just ask it directly for its sign?